File: 백준/Platinum/1090. 체커/체커.py
import sys
import math
input = sys.stdin.readline
n = int(input())

arr = []
x_arr = []
y_arr = []
for _ in range(n):
    x, y = map(int,input().split())
    arr.append((x,y))
    x_arr.append(x)
    y_arr.append(y)

s_arr = []
for x in x_arr:
    for y in y_arr:
        s_arr.append((x,y))
s_arr = list(set(s_arr))

ans = []
for i in s_arr:
    tmp = []
    for j in arr:
        tmp.append(abs(j[0] - i[0]) + abs(j[1] - i[1]))
    tmp.sort()
    ans.append(tmp)

for i in range(n):
    if i == 0:
        print(i, end =' ')
        continue
    ans2 = 1e9
    for j in ans:
        ans2 = min(ans2, sum(j[:i + 1]))
    print(ans2, end =' ')

File: 백준/Platinum/1090. 체커/체커.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;
        StringBuilder sb = new StringBuilder();
        int n = Integer.parseInt(br.readLine());
        List<int[]> list = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            st = new StringTokenizer(br.readLine());
            int x = Integer.parseInt(st.nextToken());
            int y = Integer.parseInt(st.nextToken());
            list.add(new int[]{x, y});
        }

        List<int[]> points = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                points.add(new int[]{list.get(i)[0], list.get(j)[1]});
            }
        }

        List<List<Integer>> distance = new ArrayList<>();

        for (int j = 0; j < points.size(); j++) {
            List<Integer> tmp = new ArrayList<>();
            for (int i = 0; i < n; i++) {
                tmp.add(Math.abs(points.get(j)[0] - list.get(i)[0]) + Math.abs(points.get(j)[1] - list.get(i)[1]));
            }
            Collections.sort(tmp);
            distance.add(tmp);
        }
        sb.append("0 ");

        for (int i = 2; i <= n ; i++) {
            int ans = Integer.MAX_VALUE;
            for (List<Integer> tmp: distance) {
                int sum = 0;
                for (int j = 0; j < i; j++) {
                    sum += tmp.get(j);
                }
                ans = Math.min(ans, sum);
            }
            sb.append(ans + " ");
        }
        System.out.println(sb);

    }
}
